packages/aws-cdk-lib/aws-events/lib/event-pattern.ts

```diff
@@ -109,8 +116,45 @@ export class Match implements IResolvable {
   /**
    * Matches any string that doesn't start with the given prefix.
    */
-  public static anythingButPrefix(prefix: string): string[] {
-    return this.fromObjects([{ 'anything-but': { prefix: prefix } }]);
+  public static anythingButPrefix(...values: string[]): string[] {
+    if (values.length === 0) {
+      throw new Error('anythingBut matchers must be non-empty lists');
+    }
+
+    return this.fromObjects([{ 'anything-but': { prefix: values } }]);
```

Comment:
This will always output an array also when only a single value is supplied.

To avoid changes to existing rules created with this pattern, I can add an exception to handle such cases.